src/vcf/validator.cpp

```diff
@@ -132,6 +132,65 @@ namespace ebi
         }
     }
 
+    bool is_compressed_file(std::istream &input,
+                            const std::string &source)
+    {
+        std::string message = "Input file is a compressed file ";
+
+        if (source != ebi::vcf::STDIN) {
+            boost::filesystem::path source_name(source);
+            std::string file_extension = source_name.extension().string();
+
+            if (file_extension == ".zip") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(zip)";
+                return true;
+            }
```

Comment:
this block is repeated several times. it should be extracted to another function

Reply:
I will merge these `if-else-if` statements into one. and also make a seperate function for this extension checking